### PR TITLE
fix issue with _test filename suffix in pytest testrunner

### LINKED ISSUE #740: Determine why tests are failing on PyPy
After switching to PyTest 7.0 (documented in #660) the test suite in PyPy has been failing consistently:

```
Autodetected filename timings.
No timing found for "tests/basilisp/cli_test.py"
No timing found for "tests/basilisp/list_test.py"
No timing found for "tests/basilisp/prompt_test.py"
No timing found for "tests/basilisp/stacktrace_test.lpy"
No timing found for "tests/basilisp/test_edn.lpy"
No timing found for "tests/basilisp/test_string.lpy"
Test files:
tests/basilisp/core_test.py
tests/basilisp/namespace_test.py
tests/basilisp/runtime_test.py
tests/basilisp/test_protocols.lpy
tests/basilisp/var_test.py
tests/basilisp/set_test.py
tests/basilisp/stacktrace_test.lpy
tests/basilisp/symbol_test.py
tests/basilisp/atom_test.py

.pkg: _optional_hooks> python /opt/pypy/lib/pypy3.10/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: get_requires_for_build_sdist> python /opt/pypy/lib/pypy3.10/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: get_requires_for_build_wheel> python /opt/pypy/lib/pypy3.10/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: freeze> python -m pip freeze --all
.pkg: cffi==1.15.1,greenlet==0.4.13,hpy==0.0.4,pip==23.3.1,poetry-core==1.8.1,readline==6.2.4.1,setuptools==69.0.2,wheel==0.42.0
.pkg: prepare_metadata_for_build_wheel> python /opt/pypy/lib/pypy3.10/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
.pkg: build_sdist> python /opt/pypy/lib/pypy3.10/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
pypy3: install_package> python -I -m pip install --force-reinstall --no-deps /root/project/.tox/.tmp/package/2/basilisp-0.1.0a2.tar.gz
pypy3: freeze> python -m pip freeze --all
pypy3: attrs==23.1.0,basilisp @ file:///root/project/.tox/.tmp/package/2/basilisp-0.1.0a2.tar.gz#sha256=aa6a8291c18acb4e4a1c781b7779f92c1130c85b162fe38605225f0ee9eff929,cffi==1.15.1,coverage==7.3.4,exceptiongroup==1.2.0,greenlet==0.4.13,hpy==0.0.4,immutables==0.20,iniconfig==2.0.0,packaging==23.2,pip==23.3.1,pluggy==1.3.0,prompt-toolkit==3.0.43,Pygments==2.17.2,pyrsistent==0.18.1,pytest==7.4.3,python-dateutil==2.8.2,readerwriterlock==1.0.9,readline==6.2.4.1,setuptools==69.0.2,six==1.16.0,tomli==2.0.1,typing_extensions==4.9.0,wcwidth==0.2.12,wheel==0.42.0
pypy3: commands[0]> coverage run --source=/root/project/.tox/pypy3/lib/pypy3.10/site-packages/basilisp --parallel-mode -m pytest --import-mode=importlib --junitxml=/root/project/junit/pytest/pypy3.xml tests/basilisp/core_test.py tests/basilisp/namespace_test.py tests/basilisp/runtime_test.py tests/basilisp/test_protocols.lpy tests/basilisp/var_test.py tests/basilisp/set_test.py tests/basilisp/stacktrace_test.lpy tests/basilisp/symbol_test.py tests/basilisp/atom_test.py
============================= test session starts ==============================
platform linux -- Python 3.10.13[pypy-7.3.13-final], pytest-7.4.3, pluggy-1.3.0
cachedir: .tox/pypy3/.pytest_cache
rootdir: /root/project
configfile: pyproject.toml
plugins: basilisp-0.1.0a2
collected 0 items                                                              

=============================== warnings summary ===============================
.tox/pypy3/lib/pypy3.10/site-packages/_pytest/nodes.py:649
  /root/project/.tox/pypy3/lib/pypy3.10/site-packages/_pytest/nodes.py:649: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to BasilispFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
----------- generated xml file: /root/project/junit/pytest/pypy3.xml -----------
============================== 1 warning in 0.08s ==============================
ERROR: found no collectors for /root/project/tests/basilisp/stacktrace_test.lpy

pypy3: exit 4 (272.04 seconds) /root/project> coverage run --source=/root/project/.tox/pypy3/lib/pypy3.10/site-packages/basilisp --parallel-mode -m pytest --import-mode=importlib --junitxml=/root/project/junit/pytest/pypy3.xml tests/basilisp/core_test.py tests/basilisp/namespace_test.py tests/basilisp/runtime_test.py tests/basilisp/test_protocols.lpy tests/basilisp/var_test.py tests/basilisp/set_test.py tests/basilisp/stacktrace_test.lpy tests/basilisp/symbol_test.py tests/basilisp/atom_test.py pid=746
.pkg: _exit> python /opt/pypy/lib/pypy3.10/site-packages/pyproject_api/_backend.py True poetry.core.masonry.api
  pypy3: FAIL code 4 (281.33=setup[9.29]+cmd[272.04] seconds)
  evaluation failed :( (282.17 seconds)

Exited with code exit status 4```

### PR DESCRIPTION
**Pull request by ikappaki**
Hi, 

could you please review patch to fix a break with the  recent upgrade to pytest 7 in #738 in the circle CI coverage test. Fixes #740.

The `_test` suffix was not preperly identified as a basilisp test suffix, and was failing the coverage tests for `stacktrace_test.lpy`, which was not following the same pattern as of the other test files.

I've both fixed the testrunner to properly identifying these as before the upgrade, but also renamed the  above to `test_stacktrace.lpy` for conformity with the current pattern.

Thanks

### CODE REVIEW

File: src/basilisp/contrib/pytest/testrunner.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def pytest_configure(config):
 def pytest_collect_file(file_path: Path, path, parent):
     """Primary PyTest hook to identify Basilisp test files."""
     if file_path.suffix == ".lpy":
-        if file_path.name.startswith("test_") or file_path.name.endswith("_test"):
+        if file_path.name.startswith("test_") or file_path.name.endswith("_test.lpy"):
```
Comment (chrisrink10) [inline src/basilisp/contrib/pytest/testrunner.py:32]: I think we can just check `file_path.stem.endswith("_test")`

Comment (ikappaki) [reply src/basilisp/contrib/pytest/testrunner.py:32]: I've tried this but unfortunately file_path appears to be a `string` rather than `Path`, and as such abandoned that idea, given also that the there is an `if` statement testing for the `.lpy` extension ...

Comment (ikappaki) [reply src/basilisp/contrib/pytest/testrunner.py:32]: let me try again ...

Comment (ikappaki) [reply src/basilisp/contrib/pytest/testrunner.py]: done.

**Review by chrisrink10 — APPROVED** [review]

**Review by chrisrink10 — APPROVED** [review]
